src/satosa/routing.py

```diff
@@ -134,6 +138,19 @@ def _find_registered_endpoint(self, context, modules):
 
         raise ModuleRouter.UnknownEndpoint(context.path)
 
+    def _find_backend(self, request_path):
```

Comment:
This method should only guess the backend return with `backend_guess`, so `endpoint_routing` could report better error message than `no such backend None`